src/renderer/survey/js/webview.js

```diff
@@ -0,0 +1,132 @@
+/**
+ * The preload script for the survey webview
+ *
+ * Responsible for detecting when the survey has ended - either when it has been
+ * completed, or when the user has chosen to exit it via a close button within the
+ * content.
+ *
+ * Licensed under the New BSD license. You may not use this file except in
+ * compliance with this License.
+ * The research leading to these results has received funding from the European Union's
+ * Seventh Framework Programme (FP7/2007-2013) under grant agreement no. 289016.
+ * You may obtain a copy of the License at
+ * https://github.com/GPII/universal/blob/master/LICENSE.txt
+ */
+
+"use strict";
+/**
+ * As the webview runs in a separate process than the PSP, the
+ * communication between the application and the embedded content will
+ * be asynchronous.
+ * This script will be loaded before other scripts in the guest page. It
+ * has access to all Node API (including the `ipcRenderer`). All DOM
+ * manipulations (e.g. attaching event listeners, adding new elements, etc.)
+ * need to be performed only after the DOM content has loaded.
+ */
+(function () {
+    var ipcRenderer = require("electron").ipcRenderer;
+
+    /**
+     * Checks whether a given DOM element is a 'break out' link, i.e.
+     * whether it is an anchor tag whose target is blank and which has
+     * the `flc-breakOut` class. The `flc-breakOut` class helps the
+     * application distinguish between regular links that should open in a
+     * new browser window and the 'break out' link which should also
+     * close the survey pop-up.
+     * @param target {Element} The DOM element to be checked.
+     */
+    function isBreakOutLink(target) {
```

Comment:
Please add some bare-bones namespacing to this file so that this library of functions can expand gracefully - e.g. something like var gpii = gpii || {webview: {}}
And then the contents of this file can be registered as, e.g.
gpii.webview.isBreakoutLink = function (target) {

etc.